pkg/disk/partition_table.go

```diff
@@ -160,6 +161,291 @@ func NewPartitionTable(basePT *PartitionTable, mountpoints []blueprint.Filesyste
+				vgname = fmt.Sprintf("vg%02d", idx)
+			}
+			newvg := &LVMVolumeGroup{
+				Name:        vgname,
+				Description: "created via lvm2 and osbuild",
+			}
+			for _, lv := range vg.LogicalVolumes {
+				newfs := &Filesystem{
+					Type:         fsType(lv.Type),
+					Label:        lv.Label,
+					Mountpoint:   lv.Mountpoint,
+					FSTabOptions: "defaults", // TODO: add customization
+				}
+				if _, err := newvg.CreateLogicalVolume(lv.Name, lv.MinSize, newfs); err != nil {
+					return nil, fmt.Errorf("error creating logical volume %q (%s): %w", lv.Name, lv.Mountpoint, err)
+				}
+			}
+
+			// create partition for volume group
+			newpart := Partition{
+				Type:     LVMPartitionGUID,
+				Size:     vg.MinSize,
+				Bootable: false,
+				Payload:  newvg,
+			}
+
+			pt.Partitions = append(pt.Partitions, newpart)
+		}
+	}
+
+	if customizations.Btrfs != nil {
+		for _, btrfsvol := range customizations.Btrfs.Volumes {
+			subvols := make([]BtrfsSubvolume, len(btrfsvol.Subvolumes))
+			for idx, subvol := range btrfsvol.Subvolumes {
+				newsubvol := BtrfsSubvolume{
+					Name:       subvol.Name,
+					Mountpoint: subvol.Mountpoint,
+				}
+				subvols[idx] = newsubvol
+			}
+
+			newvol := &Btrfs{
+				Subvolumes: subvols,
+			}
+
+			// create partition for btrfs volume
+			newpart := Partition{
+				Type:     FilesystemDataGUID,
+				Bootable: false,
+				Payload:  newvol,
+				Size:     btrfsvol.MinSize,
+			}
+
+			pt.Partitions = append(pt.Partitions, newpart)
+		}
+	}
+
+	// Add root partition if it wasn't created already.
+	// When adding the root partition, add it to:
+	// - The first LVM Volume Group if one exists, otherwise
+	// - The first Btrfs volume if one exists, otherwise
+	// - At the end of the plain partitions.
+	if entityPath(pt, "/") == nil {
+	PartitionLoop:
+		for _, part := range pt.Partitions {
+			switch payload := part.Payload.(type) {
+			case *LVMVolumeGroup:
+				rootfs := &Filesystem{
+					Type:         options.DefaultFSType.String(),
+					Label:        "root",
+					Mountpoint:   "/",
+					FSTabOptions: "defaults",
+				}
+				// Let the function autogenerate the name to avoid conflicts
+				// with LV names from customizations.
+				// Set the size to 0 and it will be adjusted by
+				// EnsureDirectorySizes() and relayout().
+				if _, err := payload.CreateLogicalVolume("", 0, rootfs); err != nil {
+					return nil, fmt.Errorf("error creating root logical volume: %w", err)
+				}
+				break PartitionLoop
+			case *Btrfs:
+				rootsubvol := BtrfsSubvolume{
+					Name:       "root",
+					Mountpoint: "/",
+				}
+				payload.Subvolumes = append(payload.Subvolumes, rootsubvol)
+				break PartitionLoop
+			}
+		}
+		// check again and create a plain partition if root wasn't created in lvm or btrfs
+		if entityPath(pt, "/") == nil {
+			// add a plan root partition at the end of the partition table
+			rootpart := Partition{
+				Type:     FilesystemDataGUID,
+				Bootable: false,
+				Size:     0,
```

Comment:
I don't know.  0 means autoadjust only in the case of `/` and even then it might stay 0 if there are no required sizes specified.  I think a comment is the right way to go instead of a const that could imply more.